server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb

```diff
@@ -166,6 +166,8 @@ def update(cruise_config)
 
           @pluggable_scm_service.validate(scm)
 
+          scm.clearEmptyConfigurations()
```

Comment:
yes. `optional-field` would not be null if `clearEmptyConfigurations()` doesn't happen.